custom_components/truenas/sensor.py

```diff
@@ -88,3 +92,43 @@ def _get_state(self) -> Optional[int]:
         if self.available:
             return self._disk.temperature
         return None
+
+
+class PoolSensor(TrueNASPoolEntity, TrueNASSensor, Entity):
+    _pool: Pool
+
+    def __init__(
+        self,
+        entry: ConfigEntry,
+        name: str,
+        pool: Pool,
+        coordinator: DataUpdateCoordinator,
+    ) -> None:
+        self._pool = pool
+        super().__init__(entry, name, coordinator)
+
+    @property
+    def name(self) -> str:
+        """Return the name of the pool."""
+        return f"{self._pool.name} Pool"
+
+    @property
+    def unique_id(self):
+        """Return the Unique ID of the pool."""
+        return slugify(self._pool.guid)
+
+    @property
+    def id(self) -> int:
+        """Return the ID of the pool."""
+        return self._pool.id
+
+    @property
+    def icon(self):
+        """Return an icon for the pool."""
+        return "mdi:database"
+
+    def _get_state(self):
+        """Returns the current status of the pool."""
+        if not isinstance:
+            return None
+        return self._pool.status.name
```

Comment:
Yes, the status is returned. I'm not sure if it's because of the ```Pool``` class or the API, but when ```self._pool.status``` is passed, ```PoolStatus.ONLINE``` is returned as the state, but if ```self._pool.status.name``` is passed, then just the status, i.e. ```ONLINE```, ```DEGRADED```, etc. is returned as the state.

Reply:
I can't explain the latter behavior (makes no sense to me given the [code](https://github.com/sdwilsh/hass-truenas/blob/main/custom_components/truenas/__init__.py#L205)!), but what's incorrect about the former behavior of returning `PoolStatus.ONLINE`?

Reply:
It is not incorrect, I just think it looks 'cleaner' and makes more sense (to me, anyway) for the state of the sensor to not have the prefix. After doing some digging, I did find out why this happens. The value returned from the ```fromValue``` func is bound to the func name, ```PoolStatus``` in this case. [Ref](https://docs.python.org/3.8/reference/compound_stmts.html#function-definitions). If that func returned a str it would return just the state.

Reply:
My opinion is here it should remain ```self._pool.status```, and if we want the state to just display the state, then aiotruenas needs to be updated.

Reply:
For whatever reason, every time I've read this bit of code, I read it as `self._pool.name`, which is why I was pretty confused.

I agree with you that it should be a string, and therefore `self._pool.status.name` is fine.  My bad on the confusion, as I can't believe I have, multiple times now, misread that code :(